src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            newNode.prev = null;
+            newNode.next = null;
+            head = newNode;
+            tail = newNode;
+            incrementSize();
+        } else {
+            tail.next = newNode;
+            newNode.next = null;
+            newNode.prev = tail;
+            tail = newNode;
+            incrementSize();
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = head;
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            addInStart(newNode);
+            return;
+        } else {
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
```

Comment:
you can move this to a separate method `getNodeByIndex`, there you can check if it's more efficient to iterate from the head ot from the tail